Signal/src/ViewControllers/ConversationView/Cells/OWSMessageCell.m

```diff
@@ -1168,50 +1208,41 @@ - (void)prepareForReuse
 {
     [super prepareForReuse];
 
-    [NSLayoutConstraint deactivateConstraints:self.payloadConstraints];
-    self.payloadConstraints = nil;
-    [NSLayoutConstraint deactivateConstraints:self.contentConstraints];
-    self.contentConstraints = [NSMutableArray new];
-    [NSLayoutConstraint deactivateConstraints:self.dateHeaderConstraints];
-    self.dateHeaderConstraints = nil;
-    [NSLayoutConstraint deactivateConstraints:self.footerConstraints];
-    self.footerConstraints = nil;
+    [NSLayoutConstraint deactivateConstraints:self.viewConstraints];
+    self.viewConstraints = [NSMutableArray new];
 
     self.dateHeaderLabel.text = nil;
     self.dateHeaderLabel.hidden = YES;
-    self.textView.text = nil;
-    self.textView.hidden = YES;
-    self.textView.dataDetectorTypes = UIDataDetectorTypeNone;
+    [self.bodyTextViewCached removeFromSuperview];
+    self.bodyTextViewCached.text = nil;
+    self.bodyTextViewCached.hidden = YES;
+    self.bodyTextViewCached.dataDetectorTypes = UIDataDetectorTypeNone;
     [self.failedSendBadgeView removeFromSuperview];
     self.failedSendBadgeView = nil;
-    [self.tapForMoreLabel removeFromSuperview];
-    self.tapForMoreLabel = nil;
     self.footerLabel.text = nil;
     self.footerLabel.hidden = YES;
-    self.textBubbleImageView.image = nil;
-    self.textBubbleImageView.hidden = YES;
-    self.mediaMaskingView.maskedSubview = nil;
-    self.mediaMaskingView.hideTail = NO;
-    self.mediaMaskingView.layoutMargins = UIEdgeInsetsZero;
-
-    [self.stillImageView removeFromSuperview];
-    self.stillImageView = nil;
-    [self.animatedImageView removeFromSuperview];
-    self.animatedImageView = nil;
-    [self.customView removeFromSuperview];
-    self.customView = nil;
-    [self.attachmentPointerView removeFromSuperview];
-    self.attachmentPointerView = nil;
-    [self.attachmentView removeFromSuperview];
-    self.attachmentView = nil;
-    [self.audioMessageView removeFromSuperview];
-    self.audioMessageView = nil;
-    [self.attachmentUploadView removeFromSuperview];
-    self.attachmentUploadView = nil;
+
+    self.bubbleView.hidden = YES;
+    self.bubbleView.bubbleColor = nil;
+    self.bubbleView.bubbleStrokeView = nil;
+
+    for (UIView *subview in self.bubbleView.subviews) {
+        [subview removeFromSuperview];
+    }
+
+    if (self.unloadCellContentBlock) {
+        self.unloadCellContentBlock();
```

Comment:
I don't think this is dangerous or anything, but if we're removing all the subviews, this should be redundant right?

Reply:
Yup, it should be - but let's be defensive.